lib/ansible/module_utils/gcp.py

```diff
@@ -231,7 +231,7 @@ def _validate_credentials_file(module, credentials_file, require_valid_json=True
     """
     try:
         # Try to read credentials as JSON
-        with open(credentials_file) as credentials:
+        with open(os.path.expanduser(credentials_file)) as credentials:
```

Comment:
Tried that, didn't work! How is this supposed to work? I've never heard of python doing magic on argspecs.